tests/rptest/archival/s3_client.py

```diff
@@ -292,3 +295,10 @@ def list_objects(self, bucket):
                                            Key=item['Key'],
                                            ETag=item['ETag'][1:-1],
                                            ContentLength=item['Size'])
+
+    def list_buckets(self):
+        try:
+            return self._cli.list_buckets()
+        except Exception:
+            self.logger.warn("Error listing buckets")
```

Comment:
ISTR a similar change on your other SI PR -- this should either be a logger.exception(...) call (to record some context), or don't catch it and let some higher level log the exception (e.g. if this exception will cause a test failiure)